fix: remove mdns_legacy

lidel opened the pull request:
This PR removes the ancient `mdns_legacy` implementation.

- The modern DNS-SD compatible [zeroconf implementation](https://github.com/libp2p/zeroconf#readme) (based on [new specs here](https://github.com/libp2p/specs/blob/master/discovery/mdns.md)) has been running next to the legacy one for a while (since v0.11)
  - This means we were sending twice as many packets, most likely contributing to issues like  ipfs/go-ipfs#8695
- [@libp2p/mdns](https://github.com/libp2p/js-libp2p-mdns) used by js-ipfs implements and runs both [old](https://github.com/libp2p/js-libp2p-mdns/tree/master/src/compat) and [new](https://github.com/libp2p/js-libp2p-mdns/tree/master/src), so we won't lose any interop connectivity. 

@marten-seemann @schomatis thoughts on removing old one? any concerns?

marten-seemann commented on core/node/libp2p/discovery.go:
```diff
@@ -49,12 +48,6 @@ func SetupDiscovery(useMdns bool, mdnsInterval int) func(helpers.MetricsCtx, fx.
 			if mdnsInterval == 0 {
```
You can remove this as well.

marten-seemann replied:
This is not configurable any more in the new mDNS implementation.

marten-seemann commented on docs/config.md:
```diff
@@ -108,20 +108,24 @@ config file at runtime.
     - [`Swarm.DisableBandwidthMetrics`](#swarmdisablebandwidthmetrics)
     - [`Swarm.DisableNatPortMap`](#swarmdisablenatportmap)
     - [`Swarm.EnableHolePunching`](#swarmenableholepunching)
+    - [`Swarm.EnableAutoRelay`](#swarmenableautorelay)
```
These changes seem unrelated.

marten-seemann reviewed (approved):
Let's kill it. It's been a while.
Looking forward to removing it from libp2p as well.

lidel commented:
quick cc:
- @achingbrain fysa (in case you want to remove old one from js-ipfs as well)
- @dignifiedquire  iroh is using only the new one ([from this spec](https://github.com/libp2p/specs/blob/master/discovery/mdns.md)), right?

dignifiedquire commented:
> @dignifiedquire iroh is using only the new one ([from this spec](https://github.com/libp2p/specs/blob/master/discovery/mdns.md?rgh-link-date=2022-06-20T11%3A08%3A39Z)), right?

yes, I believe that's all rust-libp2p implements at this point

lidel commented:
Removed `Discovery.MDNS.Interval` because it is no longer configurable (https://github.com/ipfs/go-ipfs/pull/9048#discussion_r906814717).

Replaced `int` with `*OptionalInteger` in the config struct so existing users are not impacted, but new installs won't have a hardcoded value anymore. 

Ready for final review / merge – my hope is to ship this in 0.14.

Jorropo reviewed (approved):
LGTM
